Automattic/jetpack
# Store the list of active plugins that uses connection in an option

Store the list of active plugins using the connection package in an option that can be synced back to wpcom.

#### Changes proposed in this Pull Request:

* Every plugin that relies in the connection package is added to an option in the database
* Keeps this option updated when plugins are added/removed
* whitelist this option name so it can be synced

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* It modifies the connection package

#### Testing instructions:

* Check the value of `jetpack_connection_active_plugins` option. It should include only jetpack
* Create a new plugin with this content:
```PHP
/** Plugin Name: Jetpack Connection Test Plugin (1) */
add_action( 'plugins_loaded', function() {
	( new Automattic\Jetpack\Config() )->ensure( 'connection', array( 'slug' => 'jetpack-connection-test-1', 'name' => 'Jetpack Connection Test Plugin (1)' ) );
}, 1 );
```
* Activate it
* Check the option value again, it should include the slug of this second plugin
* Deactivate it and check the option, it should include only jetpack again.

###### Testing sync

* Apply D43191-code
* sandbox wordpress.com
* activate and deactivate plugins in your local site
* Check remote database to see if the option was synced
